packages/core/src/components/Dialog/Dialog.jsx

```diff
@@ -0,0 +1,107 @@
+import AriaModal from 'react-aria-modal';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export const Dialog = function(props) {
+  const {
+    actions,
+    children,
+    escapeExitDisabled,
+    onExit,
+    title,
+    ...modalProps
+  } = props;
+
+  return (
+    <AriaModal
+      dialogClass="ds-c-dialog"
+      escapeExits={!escapeExitDisabled}
+      includeDefaultStyles={false}
+      onExit={onExit}
+      titleId="dialog-title"
+      underlayClass="ds-c-dialog-wrap"
+      {...modalProps}
+    >
+      <header className="ds-c-dialog__header">
```

Comment:
This needs a `div` wrapping the header, children, and footer buttons. The div should have attribute `role="document"` on it, otherwise NVDA will not let its users navigate by up and down arrow keypress. The `Dialog` will have its normal role as defined by the react-aria-dialog, but we are adding a secondary role to force NVDA into the right browsing mode.